File: 485. Max Consecutive Ones/C++/Max Consecutive Ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int> &nums) {
        int result = 0;
        int count = 0;
        for (auto num : nums) {
            if (num == 0) {
                result = max(result, count);
                count = 0;
            } else {
                count++;
            }
        }
        result = max(result, count);
        return result;
    }
};

File: 485. Max Consecutive Ones/Python/Max Consecutive Ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        count = 0
        for num in nums:
            if num == 0:
                result = max(result, count)
                count = 0
            else:
                count += 1
        result = max(result, count)
        return result
